### DISCUSSION TITLE
Pump plugin updates

### DISCUSSION THREAD

**Discussion by PlaneStuffRC** [Dan-in-CA/SIP#268] [Ideas]
I almost finished writing a new plugin for the pump module. I replaced the i2c comm with CanBus and socketcan.
Communication its reliable and there will be some updated to the code.

So far i discovered some issues with the old plugin: All there values are not updating, if values are modified in arduino, they will not be updated in the web page.
![future updates](https://user-images.githubusercontent.com/15807238/122476136-633bad00-cfce-11eb-980e-15d72197cf5c.png)
There are a few things that i would like to have in the new plugin, but in have no experience with web.py :)
A few boxes with some values like:canbus address,temperature,humidity etc.
This is the [repo ](https://github.com/adrianbostan/sip_plugins/blob/main/canbus_pump_control/canbus_pump_control.py)of the new plugin
Also a box that became green when the module its connected and its working

**Comment by Dan-in-CA** [comment]
The repo link is not working. -- Never mind I found it. -- Nope it was the Pump control module.  
still have not found the Python code.

**Reply by PlaneStuffRC** [reply]
I sent an invite now, i forgot it was private

**Comment by Dan-in-CA** [comment]
OK got it.

**Reply by Dan-in-CA** [reply]
I have the Python file but I will also need the template (HTML) file to get things working.

**Reply by PlaneStuffRC** [reply]
Its the same HTML as for i2c pump

**Reply by PlaneStuffRC** [reply]
Some configs will be needed in order to enable the can interface:
https://www.beyondlogic.org/adding-can-controller-area-network-to-the-raspberry-pi/

Maybe its possible to do this from the python plugin

**Reply by Dan-in-CA** [reply]
It should be possible to do the configs from the plugin.
Python can run Linux commands so the plugin can check if a driver is loaded and edit the config files if needed.
